test-framework/test-suites/unit/tests/pylib/stack/test_pylib_stack_prober.py

```diff
@@ -0,0 +1,66 @@
+	['RHEL', '8.0.0', 'redhat8', 'x86_64', 'redhat', {'rhel-8.0-x86_64-dvd.iso.treeinfo': '.treeinfo', 'rhel-8.0-x86_64-dvd.iso.discinfo': '.discinfo'}],
+	['Fedora', '30', 'fedora30', 'x86_64', 'redhat', {'Fedora-Server-dvd-x86_64-30-1.2.iso.treeinfo': '.treeinfo', 'Fedora-Server-dvd-x86_64-30-1.2.iso.discinfo': '.discinfo'}],
+	['SLES', '11.3', '1.138', 'x86_64', 'sles', {'SLES-11-SP3-DVD-x86_64-GM-DVD1.iso.content': 'content'}],
+	['SLE-SDK', '11.3', '1.69', 'x86_64', 'sles', {'SLE-11-SP3-SDK-DVD-x86_64-GM-DVD1.iso.content': 'content'}],
+	['SLE-SDK', '12', 'sp2', 'x86_64', 'sles', {'SLE-12-SP2-SDK-DVD-x86_64-GM-DVD1.iso.content': 'content'}],
+	['SLES', '12', 'sp3', 'x86_64', 'sles', {'SLE-12-SP3-Server-DVD-x86_64-GM-DVD1.iso.content': 'content'}],
+	['SLES', '15', 'sles15', 'x86_64', 'sles', {'SLE-15-Installer-DVD-x86_64-GM-DVD1.iso.treeinfo': '.treeinfo'}],
+	['Ubuntu', '18.04.3', 'ubuntu18.04', 'x86_64', 'Ubuntu', {'ubuntu-18.04.3-desktop-amd64.iso.disk.info': '.disk/info'}],
+	['Ubuntu-Server', '18.04.3', 'ubuntu18.04', 'x86_64', 'Ubuntu', {'ubuntu-18.04.3-live-server-amd64.iso.disk.info': '.disk/info'}],
+	['Ubuntu', '19.04', 'ubuntu19.04', 'x86_64', 'Ubuntu', {'ubuntu-19.04-desktop-amd64.iso.disk.info': '.disk/info'}],
+	['Ubuntu-Server', '19.04', 'ubuntu19.04', 'x86_64', 'Ubuntu', {'ubuntu-19.04-live-server-amd64.iso.disk.info': '.disk/info'}],
+]
+
+class TestProbepal:
+
+	@pytest.mark.parametrize('name, version, release, arch, distro_family, filemap', PALLET_DATA)
+	def test_prober(self, test_file, name, version, release, arch, distro_family, filemap):
+		prober = pal.Prober()
+
+		with tempfile.TemporaryDirectory() as paldir:
+			for input_file, dest in filemap.items():
+				pathlib.Path(f'{paldir}/{dest}').parent.mkdir(parents=True, exist_ok=True)
+				shutil.copyfile(test_file(f'pallet_artifacts/{input_file}'), f'{paldir}/{dest}')
+			pal_map = prober.find_pallets(paldir)
+			assert pal_map[paldir] == pal.PalletInfo(name, version, release, arch, distro_family)
+
+	def test_prober_multi_args(self, test_file):
+		paldirs = []
+		palinfos = []
+
+		*pallet_data, filemap = PALLET_DATA[0]
+		paldir = tempfile.TemporaryDirectory()
+		for input_file, dest in filemap.items():
+			pathlib.Path(f'{paldir.name}/{dest}').parent.mkdir(parents=True, exist_ok=True)
+			shutil.copyfile(test_file(f'pallet_artifacts/{input_file}'), f'{paldir.name}/{dest}')
+		paldirs.append(paldir.name)
+		palinfos.append(pal.PalletInfo(*pallet_data))
+
+		*pallet_data, filemap = PALLET_DATA[1]
+		paldir2 = tempfile.TemporaryDirectory()
+		for input_file, dest in filemap.items():
+			pathlib.Path(f'{paldir2.name}/{dest}').parent.mkdir(parents=True, exist_ok=True)
+			shutil.copyfile(test_file(f'pallet_artifacts/{input_file}'), f'{paldir2.name}/{dest}')
+		paldirs.append(paldir2.name)
+		palinfos.append(pal.PalletInfo(*pallet_data))
+
+		prober = pal.Prober()
+		pal_map = prober.find_pallets(*paldirs)
+		assert set(pal_map.values()) == set(palinfos)
```

Comment:
Why does this need to make these lists unique in order to compare them?

Reply:
Because two unsorted lists aren't equal, and sorting `pallet_info` would require adding further methods to them which would be meaningless.

The only other alternative I see is passing `sorted()` a key function:

```
l = lambda o: (o.name, o.version, o.release, o.arch, o.distro_family)
sorted(pal_map.values(), key=l) == sorted(palinfos, key=l)
```

... but that feels weird, too...

The only risk here is if `PALLET_DATA[0]` and `PALLET_DATA[1]` are probed incorrectly in the exact opposite way.

Reply:
Using set is the simplest way I can think of to compare two unsorted lists easily. I vote for keeping the test code as simple as possible.

Reply:
Although, if palinfos is a list of named tuples, you should be able to just sort them without a key. Namedtuple is just a tuple, which will sort in field order.

Reply:
When I see `set` I think "removing duplicate elements (uniqueifying)" not "sorting a list". I'd be happy with a comment (or docstring!) mentioning why the `set` is used here, especially if it's being used for sorting instead of using `sort`.

As an aside, do data classes not give you some sort of default sorting? Is there some dunder method to define what `>, >=, <, <=` does that maybe should be implemented instead?

Reply:
I'm pretty sure this would work, if both variables are lists of the same namedtuple:
```
assert sorted(pal_map.values()) == sorted(palinfos)
```